### ISSUE TITLE
HttpServer some requests timing out

### ISSUE THREAD

**Issue by bchah** [AirenSoft/OvenMediaEngine#360]
**Describe the bug**
Starting from the commits yesterday (ecae300 forward), the HttpServer seems to lock up for 1 minute almost exactly when it gets a few requests. Specifically I am seeing this with the thumbnail publisher.

**To Reproduce**
Send some repeated requests to the HttpServer with some streams active, for example in my case I am retrieving the updated thumbnail every 300ms. This has worked well for a while but suddenly the logfile is showing a clue like the example below. I tried adjusting the session and stream load balancing values but the issue still happened eventually.

**Logs**

This is part of a 50MB log file with many other tests. So to start, I highlighted just where the delay happened. Almost exactly 1 minute, very strange - was there something added recently to the code which could cause the HttpServer to sleep for 1 minute?

```
[2021-05-06 19:51:22.246] I [SPThumbnailPubl:1173] HttpServer | http_server.cpp:343  | The HTTP client(1.2.3.4:53692) has been disconnected from 0.0.0.0:20081 (200)
[2021-05-06 19:51:22.553] I [SPThumbnailPubl:1174] HttpServer | http_server.cpp:263  | Client(<ClientSocket: 0x7f9500390d00, #70, Connected, TCP, Nonblocking, 1.2.3.4:53694>) is connected on 0.0.0.0:20081
[2021-05-06 19:51:22.561] I [SPThumbnailPubl:1174] HttpServer | http_server.cpp:218  | Client(<ClientSocket: 0x7f9500390d00, #70, Connected, TCP, Nonblocking, 1.2.3.4:53694>) is requested uri: [https://myserver.net:20081/live/helloWorld_preview/thumb.jpg]
[2021-05-06 19:51:22.562] I [SPThumbnailPubl:1174] HttpServer | http_server.cpp:343  | The HTTP client(1.2.3.4:53694) has been disconnected from 0.0.0.0:20081 (200)
[2021-05-06 19:51:22.565] I [SPThumbnailPubl:1171] HttpServer | http_server.cpp:263  | Client(<ClientSocket: 0x7f9500391100, #70, Connected, TCP, Nonblocking, 1.2.3.4:53696>) is connected on 0.0.0.0:20081
[2021-05-06 19:51:22.570] I [SPThumbnailPubl:1171] HttpServer | http_server.cpp:218  | Client(<ClientSocket: 0x7f9500391100, #70, Connected, TCP, Nonblocking, 1.2.3.4:53696>) is requested uri: [https://myserver.net:20081/live/helloWorld_preview/thumb.jpg]
[2021-05-06 ****19:51:22.571****] I [SPThumbnailPubl:1171] HttpServer | http_server.cpp:343  | The HTTP client(1.2.3.4:53696) has been disconnected from 0.0.0.0:20081 (200)
[2021-05-06 ****19:52:22.636****] I [SPThumbnailPubl:1172] HttpServer | http_server.cpp:263  | Client(<ClientSocket: 0x7f9500391710, #70, Connected, TCP, Nonblocking, 1.2.3.4:53698>) is connected on 0.0.0.0:20081
[2021-05-06 19:52:22.637] I [SPThumbnailPubl:1172] HttpServer | http_server.cpp:343  | The HTTP client(1.2.3.4:53698) has been disconnected from 0.0.0.0:20081 (200)
[2021-05-06 19:52:22.640] I [SPThumbnailPubl:1173] HttpServer | http_server.cpp:263  | Client(<ClientSocket: 0x7f9500391980, #70, Connected, TCP, Nonblocking, 1.2.3.4:53700>) is connected on 0.0.0.0:20081
[2021-05-06 19:52:22.644] I [SPThumbnailPubl:1173] HttpServer | http_server.cpp:218  | Client(<ClientSocket: 0x7f9500391980, #70, Connected, TCP, Nonblocking, 1.2.3.4:53700>) is requested uri: [https://myserver.net:20081/live/bec56a0978401_preview/thumb.jpg]
[2021-05-06 19:52:22.644] I [SPThumbnailPubl:1173] HttpServer | http_server.cpp:343  | The HTTP client(1.2.3.4:53700) has been disconnected from 0.0.0.0:20081 (200)
```

**Server (please complete the following information):**
- Ubuntu 18.0.4 running commit from today f8ff9af

**Comment by bchah** [comment]
I am doing some more testing and can confirm it is _only_ the HttpServer component that hangs. The rest of OME continues working. I hope this helps as a clue!

**Comment by dimiden** [comment]
@bchah 
Thank you very much!
Recently, I developed ```HttpClient``` for the "Signed Callback" function, so I refactored a lot.
I expect it to be an issue related to this, and I will take more various tests based on what you said.

**Comment by bchah** [comment]
@dimiden I did some more testing and have updated the title of this thread to 'some requests timing out'.

I'm sorry for not realizing this earlier but the 1 minute timeout is happening in my application which uses PHP, and it has a 60 second socket timeout duration which is timing out when using the PHP function header(_<OME HttpServer endpoint>_). So it is not a clue itself, but still it should be worth researching why some requests are not being served and timing out.

For any **PHP** users who are having issues related to the HttpServer, you can set the socket timeout duration to 1s as a workaround in your code. Probably good to keep this interval low anyways since all requests to OME should be responded in less than 1s:

`ini_set('default_socket_timeout', 1);`

**Comment by dimiden** [comment]
@bchah
Thank you for your detailed explanation. I just solved the problem and committed it to a2d64db, and I hope it works well.

**Comment by dimiden** [comment]
@bchah 
I'm glad that the code works well. 
The error message is an error message that occurs in normal circumstances, but it can be confusing, so I modified it to prevent it from being displayed.
